### PR TITLE
Filters, Mark Unread/Read/Favourites FF removals + code and tests clean up

### PR DESCRIPTION
**Pull request by Velin92**
The title says it all.
We also removed the user suggestions, since it's a feature that was not completed for quite some time.

I have also run periphery to remove some unused code

### CODE REVIEW

File: ElementX/Sources/Services/Room/RoomSummary/RoomSummaryProvider.swift [hunk 1]
```diff
@@ -24,9 +24,7 @@ class RoomSummaryProvider: RoomSummaryProviderProtocol {
     private let name: String
     private let shouldUpdateVisibleRange: Bool
     private let notificationSettings: NotificationSettingsProxyProtocol
-    private let backgroundTaskService: BackgroundTaskServiceProtocol
```
Comment (Velin92) [inline ElementX/Sources/Services/Room/RoomSummary/RoomSummaryProvider.swift:27]: Did this thing have any reason to be here at all? Periphery spotted it has assigned but unused.

File: ElementX/Sources/Services/Room/RoomMember/RoomMemberProxy.swift [hunk 2]
```diff
@@ -18,16 +18,9 @@ import Foundation
 import MatrixRustSDK
 
 final class RoomMemberProxy: RoomMemberProxyProtocol {
-    private let backgroundTaskService: BackgroundTaskServiceProtocol
```
Comment (Velin92) [inline ElementX/Sources/Services/Room/RoomMember/RoomMemberProxy.swift:21]: same comment as the other one

File: ElementX/Sources/Services/Room/RoomMember/RoomMemberProxy.swift
Comment (pixlwave) [inline ElementX/Sources/Services/Room/RoomMember/RoomMemberProxy.swift:1]: My bad, forgot about these 🙈

**Review by pixlwave — APPROVED** [review]
LGTM 👍

File: ElementX/Sources/Services/Room/RoomMember/RoomMemberProxy.swift (hunk 2 above)
Comment (pixlwave) [reply ElementX/Sources/Services/Room/RoomMember/RoomMemberProxy.swift:21]: Yeah this one did, it was for ignoring members when the methods were on the proxy.

File: ElementX/Sources/Services/Room/RoomSummary/RoomSummaryProvider.swift (hunk 1 above)
Comment (pixlwave) [reply ElementX/Sources/Services/Room/RoomSummary/RoomSummaryProvider.swift:27]: Possibly for back pagination before that was moved to the timeline?

File: ElementX/Sources/Services/Room/RoomMember/RoomMemberProxy.swift
Comment (Velin92) [reply ElementX/Sources/Services/Room/RoomMember/RoomMemberProxy.swift:1]: np, periphery is always here to guide us towards cleaner code

**Pull request merged**